infra/nightly-resources/index.js

```diff
@@ -1,46 +1,202 @@
-    if(!groupedBy[obj.benchmark]) {
-      groupedBy[obj.benchmark] = [];
+async function getPreviousRuns() {
+    const req = await fetch("../");
+    const html = await req.text();
+
+    const parser = new DOMParser();
+    const htmlDoc = parser.parseFromString(html, 'text/html');
+
+    const allLinks = htmlDoc.getElementsByTagName("a");
+    
+    // map allLinks into an objects of the shape:
+    // {branch: <git branch:string>, commit: <git commit:string>, date: <unix timestamp:int>, url: <absolute url to nightly page:string>}
+    const comparisons = [];
+    for (let i = 1; i < allLinks.length; i++) {
+        const hrefText = allLinks[i].getAttribute("href");
+
+        const [date, _, branch, commit] = hrefText.split("%3A");
+
+        const run = {
+            branch: branch,
+            // since commit is the last item in the link it has a trailing slash, so we slice it off
+            commit: commit.slice(0, -1),
+            // type coerce a unix timestamp that is a string into an int with a `+`
+            date: +date,
+            // The file server/DOM parser constructs the absolute URL (of a relative URL) based on where we're requesting from,
+            // and since we're requesting from a sibling page, it mangles the link to look something like:
+            //   "/reports/eggcc/<current report>/<report requested>" 
+            // which is a nonsensical url. We can reconstruct the URL with the href attribute which we do here
+            url: `https://nightly.cs.washington.edu/reports/eggcc/${hrefText}`, 
+        }
+
+        comparisons.push(run);
+    }
+
+    comparisons.sort((l, r) => {
+        if (l.date < r.date) {return -1; }
+        if (l.date > r.date) {return 1; }
+        return 0;
+    });
+
+    return comparisons;
+}
+
+async function buildNightlyDropdown(element, previousRuns, initialIdx) {
+    const select = document.getElementById(element);
+    
+    const formatRun = (run) => `${run.branch} - ${run.commit}`
+    
+    previousRuns.forEach((nightly) => {
+        console.log(nightly);
+
+        const option = document.createElement("option");
+        option.innerText = formatRun(nightly);
+        select.appendChild(option);
+    });
+    
+    select.onchange = () => {
+        const compareTo = previousRuns[select.selectedIndex];
+        loadBenchmarks(compareTo);
+    }
+
+    select.selectedIndex = initialIdx;
+    select.value = formatRun(previousRuns[initialIdx]);
+    select.onchange();
+}
+
+// findBenchToCompare will find the last benchmark run on the main branch that is not itself
+function findBenchToCompareIdx(benchRuns) {
+    // Determine what benchmark run we are based on the browser's URL
+    // This is likely the best way to do this without embedding a bunch of data into our profile.js file
+    // or our profile.json file, which although tempting, is not backwards compatible
+    const path = window.location.pathname;
+    const parts = path.split("/");
```

Comment:
is there not a nice javascript function to get the end of the URL?

Reply:
Nope, `pathname` is all I could find on the interwebs. Weird huh?

Reply:
dang